### ISSUE TITLE
🐛 take photo returns photo file with width and height flipped

### ISSUE THREAD

**Issue by kevmai96** [cuvent/react-native-vision-camera#1401]
### What were you trying to do?

I have been struggling with cropping the image file after 

let { path, width, height } = await camera.current.takePhoto({
      qualityPrioritization: "speed",
      skipMetadata: true,
      flash: "off"
    })

### What happened instead?

The width and height is flipped. width is height and height is width.

### Device

iPhone 11

### VisionCamera Version

^2.15.2

### Additional information

- [X] I am using Expo
- [X] I have read the [Troubleshooting Guide](https://mrousavy.github.io/react-native-vision-camera/docs/guides/troubleshooting)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/mrousavy/react-native-vision-camera/blob/main/CODE_OF_CONDUCT.md)
- [X] I searched for [similar issues in this repository](https://github.com/mrousavy/react-native-vision-camera/issues) and found none.

**Comment by lucipacurar** [comment]
I have the same problem and it triggers a lot of other issues: we're doing some auto cropping and that is broken because of this, and in some parts of our backend the images are flipped when adding them to a PDF.

**Comment by kevmai96** [comment]
and also, i have two android phones, the front taking photo is flipped in one and not the other

**Comment by lucipacurar** [comment]
As a workaround I'm using https://github.com/bamlab/react-native-image-resizer to resize the image to the same size and it fixes the orientation issues.

**Comment by kevmai96** [comment]
const photo = await camera.current.takePhoto({
      qualityPrioritization: "speed",
      skipMetadata: true,
      flash: "off",
    })
let path = photo.path, width = photo.width, height = photo.height
    let uri = Platform.OS == "ios" ? path : `file://${path}`

ImageResizer.createResizedImage(
		uri,
		width,
		height,
		"JPEG",
		100,
		0,
		undefined,
		false
)
	.then(async(response) => {
})
	  .catch((err) => {
	    // Oops, something went wrong. Check that the filename is correct and
	    // inspect err to get more details.
	    console.log(err, path)
	  });


ios logs: [Error: Image uri invalid.] /private/var/mobile/Containers/Data/Application/324E4E06-EF5C-4F5B-A4EF-5FD7B89E607D/tmp/ReactNative/EBDB1F6D-43A5-404B-875D-DF68C7FFB64D.jpeg

android: doesn't fix orientation

any help would be great thanks:)

**Comment by lucipacurar** [comment]
We used destination directory as well. 

```
const compressedImage = await ImageResizer.createResizedImage(
					`file://${photo.path}`,
					photo.height, // width
					photo.width, // height
					'JPEG',
					80, // compression
					0, // rotation
					destinationDirectory,
				);
```

The weird thing is that I didn't find a device for which the sizes are properly.

**Comment by kevmai96** [comment]
When I set destination directory to undefined, it works.

This is what I did before, 
let destinationDirectory = FileSystem.documentDirectory + "/dsjfkldsfjsdlkjf.jpeg"
outputs: [Error: Image uri invalid.]

maybe the image uri is private....

**Comment by lucipacurar** [comment]
We're extracting the directory from `photo.path`, which I think it's a temp directory.

**Comment by kevmai96** [comment]
So you're using the same directory as the original?

**Comment by lucipacurar** [comment]
Yes, the same directory. We just extract it from `photo.path`:
```TypeScript
const photoPathArray = photo.path.split('/');
				photoPathArray.pop();
```

**Comment by kevmai96** [comment]
Found something better:)

let flipImage = false

if (photo.height > photo.width) {
flipImage = true
}

**Comment by kevmai96** [comment]
Hey there, something weird I found,

(devices.front) && (
	            <Camera
	              style={{ height: '100%', width: '100%' }}
	              ref={camera}
	              device={devices.front}
	              isActive photo video audio
	            />
	          )

When I take a photo at first, it captures. But when I record a video and then try taking a photo again, it doesn't capture.......

So I had to do this for now
(devices.front) && (
	            <Camera
	              style={{ height: '100%', width: '100%' }}
	              ref={camera}
	              device={devices.front}
	              isActive photo video={recording} audio={recording}
	            />
	          )

'recording' is set to false as default, when the user press recording and before it starts, I would set recording to true and put a 0.5 seconds timeout before starting the recording process, after I press stop recording, then I set the recording state back to false.

I hope you understand what I'm saying.

Cheers:)

**Comment by EwaGuziejko** [comment]
> As a workaround I'm using https://github.com/bamlab/react-native-image-resizer to resize the image to the same size and it fixes the orientation issues.

I tried the same but it messes up with photo dimensions :/ it didn't happen for you lucipacurar?

**Comment by lucipacurar** [comment]
No we didn't have problems yet with the dimensions. We tested on our test lab devices and we didn't receive any weird customer reports yet. But please have a second look at the code above where we switch the width for the height and vice versa.

**Comment by myselfuser1** [comment]
Try this https://www.youtube.com/playlist?list=PLQhQEGkwKZUrempLnmxjt7ZCZJu1W3p2i

**Comment by mrousavy** [comment]
Closing as this is a stale issue - this might have been fixed with the full rewrite in VisionCamera V3 (🥳) - if not, please create a new issue.

If your issue has been fixed, consider [sponsoring me on GitHub](https://github.com/sponsors/mrousavy) to say thanks 💖

JFYI; Camera sensor orientation is 90deg, maybe thats why it's flipped.

**Comment by mrousavy** [comment]
Hey - I'm tracking Orientation in this feature request/issue now: https://github.com/mrousavy/react-native-vision-camera/issues/1891 ✨ 

Make sure to upvote or sponsor to support this feature, and leave a comment if you have any additional thoughts/ideas.
